gateway/src/node/client_handling/websocket/listener.rs

```diff
@@ -64,9 +65,9 @@ where
                                 active_clients_store.clone(),
                                 self.shared_state.clone(),
                                 remote_addr,
+                                shutdown,
                             );
-                            let shutdown = shutdown.clone().named(format!("ClientConnectionHandler_{remote_addr}"));
-                            tokio::spawn(async move { handle.start_handling(shutdown).await });
+                            tokio::spawn(async move { handle.start_handling().await });
```

Comment:
Nit: not certain but you may probably give it `handle.start_handling()` without await